<a href="https://colab.research.google.com/github/imymemineyay/Study_Python/blob/main/Python_dojang_(Unit5_11).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Unit 5. 숫자 계산하기**

5.1 연산자 (정수)

In [16]:
# 정수의 덧셈
print(1+1)

# 정수의 뺄셈
print(1-1)

# 정수의 곱셈
print(5*17)

# 정수의 나눗셈 : 나눗셈 후 완전히 나누어 떨어져도 실수형으로 나타내는 연산자
print(5/2)

# 정수의 나눗셈(//) : 나눗셈 후 소수점 이하를 버리는 연산자
print(5//2)

# 정수의 나눗셈(%) : 나눗셈 후 나머지를 구하는 연산자
print(5%2)

# 정수의 제곱
print(4**2)


2
0
85
2.5
2
1
16


5.1 연산자 (실수)

In [17]:
# 실수의 덧셈
print(3.5+2.6)

# 실수의 뺄셈
print(4.5-2.2)

# 실수의 곱셈
print(2.1*6.3)

# 실수의 나눗셈
print(5.2/2.2)

# 실수와 정수 계산
print(1.2+3)


6.1
2.3
13.23
2.3636363636363633
4.2


5.1 괄호 사용하기

In [18]:
print(35 + 1 * 2)
print((35 + 1) * 2)

37
72


5.2 값을 정수로 만들기
- 소수점 이하 삭제

In [6]:
print(int(3.3))
print(int(5/2))
print(int('10'))

3
2
10


5.3 값을 실수로 만들기

In [13]:
print(float(5))
print(float(1+2))
print(float('5.4'))

5.0
3.0
5.4


5.4 객체의 자료형 확인
- 숫자 = 객체 = 클래스

In [14]:
print(type(10))
print(type(3.5))

<class 'int'>
<class 'float'>


5.5 divmod()함수 : (몫, 나머지) 출력

In [8]:
quotient, remainder = divmod(5,2) # 튜플 형태로 결과값 출력 그리고 unpacking
print(quotient,remainder)

2 1


참고) 복소수 
- 실수부와 허수부(j)

In [15]:
print(1.2+1.3j)
print(complex(1.2,1.3))

(1.2+1.3j)
(1.2+1.3j)


연습문제)

L이라는 게임에서 "왜곡"이라는 스킬이 AP * 0.6 + 225의 피해를 입힙니다. 참고로 이 게임에서 AP(Ability Power, 주문력)는 마법 능력치를 뜻합니다. 다음 소스 코드를 완성하여 스킬의 피해량이 출력되게 만드세요.

AP: 102

In [19]:
print(102*0.6+225)

286.2


## **Unit 6. 변수와 입력 사용하기**

변수 규칙
- 영문 문자와 숫자를 사용할 수 있습니다.
- 대소문자를 구분합니다.
- 문자부터 시작해야 하며 숫자부터 시작하면 안 됩니다.
- _(밑줄 문자)로 시작할 수 있습니다.
- 특수 문자(+, -, *, /, $, @, &, % 등)는 사용할 수 없습니다.
- 파이썬의 키워드(if, for, while, and, or 등)는 사용할 수 없습니다.

In [20]:
a = 10
a

10

6.1 변수의 자료형 알아내기

In [21]:
type(a)

int

참고) In 프로그래밍 언어 <br>
= 는 변수에 값을 할당한다는 의미 <br>
연산자 == 은 수학의 등호와 같은 역할

6.2 변수 여러 개를 한 번에 만들기 <br>
※ 변수와 값의 개수를 맞춰야 함

In [22]:
x, y, z = 59,45,20
print(x)
print(y)
print(z)
print(x,y,z)

59
45
20
59 45 20


In [23]:
x, y = 10, 30
x,y = y, x
print(x,y)

30 10


6.3 변수 삭제하기

In [24]:
del x
x

NameError: ignored

6.4 빈 변수 만들기 

In [26]:
x = None
print(x) # None : 아무것도 없는 상태를 나타내는 자료형

None


6.5 산술 연산 후 할당 연산자 사용하기

In [31]:
a = 10
print(a + 20)
a = a + 20 # 변수 한 개에서 값의 변화를 계속 유지하려면 계산 결과를 다시 변수에 저장
print(a)
b = 20
b += 10
print(b) # 산술 연산자 앞에 =(할당 연산자)를 붙이면 연산 결과를 변수에 저장함
         # print(b += 10) 은 invalid syntax 이유는 print(b = b + 10)은 불가능하기 때문

30
30
30


6.6 부호 붙이기

In [32]:
x = -10
print(+x)
print(-x)

-10
10


6.7 input함수 사용

In [40]:
h = input('첫 번째 숫자를 입력하세요 : ')
i = input('두 번째 숫자를 입력하세요 : ')

print(h+i) # input함수는 문자열 형태로 출력
print(type(h+i))

첫 번째 숫자를 입력하세요 : 5
두 번째 숫자를 입력하세요 : 7
57
<class 'str'>


6.8 입력 값을 정수로 변환

In [41]:
a = int(input('첫 번째 숫자를 입력하세요: '))  
b = int(input('두 번째 숫자를 입력하세요: '))
print(a + b)

첫 번째 숫자를 입력하세요: 20
두 번째 숫자를 입력하세요: 50
70


6.9 입력 값을 변수 두 개에 저장 ❤❤
- 변수1, 변수2 = input().split()
- 변수1, 변수2 = input().split('기준문자열')
- 변수1, 변수2 = input('문자열').split()
- 변수1, 변수2 = input('문자열').split('기준문자열')


In [45]:
a, b = input('문자열 두 개를 입력하세요: ').split() # 공백을 기준으로 분리
print(a)
print(b)

문자열 두 개를 입력하세요: Hello, Python
Hello,
Python


In [47]:
a, b = input('숫자 두 개를 입력하세요: ').split()    # 입력받은 값을 공백을 기준으로 분리
a = int(a)    # 변수를 정수로 변환한 뒤 다시 저장
b = int(b)    # 변수를 정수로 변환한 뒤 다시 저장
 
print(a + b)

숫자 두 개를 입력하세요: 10 20
30


6.10 map을 사용하여 정수로 변환하기
- 변수1, 변수2 = map(int, input().split())
- 변수1, 변수2 = map(int, input().split('기준문자열'))
- 변수1, 변수2 = map(int, input('문자열').split())
- 변수1, 변수2 = map(int, input('문자열').split('기준문자열'))

In [49]:
a, b = map(int, input('숫자 두 개를 입력하세요: ').split(',')) # 입력받은 값을 콤마를 기준으로 분리
 
print(a + b)

숫자 두 개를 입력하세요: 5,50
55


연습문제) <br>
표준 입력으로 국어, 영어, 수학, 과학 점수가 입력됩니다. 평균 점수를 출력하는 프로그램을 만드세요(input에서 안내 문자열은 출력하지 않아야 합니다). 단, 평균 점수를 출력할 때는 소수점 이하 자리는 버립니다(정수로 출력). <br>
입력 (83 92 87 90) 결과 (88) 


In [58]:
a, b, c, d = map(int,input().split())
print((a+b+c+d)//4)

83 92 87 90
88


## **Unit 7. 출력 방법 알아보기**

7.1 sep활용
- 현재 print함수 내의 내용에 영향을 줌

In [64]:
print('Hello', 'Python', sep='_')    # 값 사이에 문자 넣는 법
print('Doby', 'is', 'free', sep='\n') # 다음 줄 출력
print('1\n2') 
print('Bowon is', 'Bowon', sep='\t') # \t: 여러 칸을 띄움 
print('Hi','there', sep='\\') # \\: \문자 자체 출력

Hello_Python
Doby
is
free
1
2
Bowon is	Bowon
Hi\there


7.2 end활용
- print함수가 끝난 뒤, 그 다음에 오는 print함수에 영향을 줌

In [67]:
print(1, end='')
print(2, end='')
print(3) # end에 빈 문자열을 지정하면 세 줄로 출력되지 않고 한 줄로 붙어서 출력

print(1, end=' ')
print(2, end=' ')  # print함수는 \n을 내포하고 있음 
print(3, end=' ')  # end에 공백 한 칸 지정

123
1 2 3 

연습문제) <br>
표준 입력으로 년, 월, 일, 시, 분, 초가 입력됩니다. 다음 소스 코드를 완성하여 입력된 날짜와 시간을 년-월-일T시:분:초 형식으로 출력되게 만드세요.

입력 (1999 12 31 10 37 21)
결과 (1999-12-31T10: 37 : 21)

In [69]:
year,month,day,hour,minute,second = input().split()
print(year,month,day,sep='-',end='T')
print(hour,minute,second,sep=':')

1999 12 31 10 37 21
1999-12-31T10:37:21


## **Unit 8. 불과 비교, 논리 연산자 알아보기**

8.1 비교 연산자 (숫자)

In [74]:
print(3 > 1) # 크기 비교
print(10==10) # 같은지 비교 (equal)
print(10 != 10) # 다른지 비교 (not equal) 

True
True
False


8.2 비교 연산자 (문자)

In [75]:
print('Python' == 'Python')
print('Python' == 'python')
print('Python' != 'python')

True
False
True


8.3 is와 == 비교

In [78]:
print(1 == 1.0) # 값은 같음
print(1 is 1.0) # is는 객체를 판별 (id로 객체의 고유한 값 확인 가능)
print(1 is not 1.0) # is, is not은 클래스로 객체를 만든 뒤 객체가 서로 같은지 비교할 때 사용

a = -5
print(a is -5)

a = -6
print(a is -6) # 변수 a가 있는 상태에서 다른 값을 할당하면 메모리 주소 변경 가능성 있음
print(a == -6) # 값을 비교할 때는 is가 아닌 비교 연산자를 사용해야 함  


True
False
True
True
False
True


<>:2: SyntaxWarning: "is" with a literal. Did you mean "=="?
<>:3: SyntaxWarning: "is not" with a literal. Did you mean "!="?
<>:6: SyntaxWarning: "is" with a literal. Did you mean "=="?
<>:9: SyntaxWarning: "is" with a literal. Did you mean "=="?
<>:2: SyntaxWarning: "is" with a literal. Did you mean "=="?
<>:3: SyntaxWarning: "is not" with a literal. Did you mean "!="?
<>:6: SyntaxWarning: "is" with a literal. Did you mean "=="?
<>:9: SyntaxWarning: "is" with a literal. Did you mean "=="?
<ipython-input-78-3a8474cce0ec>:2: SyntaxWarning: "is" with a literal. Did you mean "=="?
  print(1 is 1.0) # is는 객체를 판별 (id로 객체의 고유한 값 확인 가능)
<ipython-input-78-3a8474cce0ec>:3: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  print(1 is not 1.0) # is, is not은 클래스로 객체를 만든 뒤 객체가 서로 같은지 비교할 때 사용
<ipython-input-78-3a8474cce0ec>:6: SyntaxWarning: "is" with a literal. Did you mean "=="?
  print(a is -5)
<ipython-input-78-3a8474cce0ec>:9: SyntaxWarning: "is" with a literal. Did you mean "=="?


8.4 논리 연산자

In [79]:
print(True and True)
print(False and True)
print(False or False)
print(True or False)
print(not True)
print(not True and False or not False)

True
False
False
True
False
True


8.5 논리 연산자와 비교 연산자 사용

In [80]:
10 == 10 and 10 != 5

True

8.6 정수, 실수, 문자열을 불로 만들기

In [81]:
print(bool(1)) # 정수 1 == True
print(bool(0)) # 정수 0 == False
print(bool(1.3)) # 숫자 == True
print(bool('False')) # 문자 == True
print(bool()) # 빈 값 == False

True
False
True
True
False


8.7 단락 평가 
- 첫 번째 값만으로 결과가 확실할 때 두 번째 값은 확인하지 않는 방법

In [83]:
print(False and True)
print(True or False)

False
True


- 파이썬에서 논리 연산자는 마지막으로 단락 평가를 실시한 값을 그대로 반환
- 논리 연산자는 무조건 불을 반환하지 않음

In [88]:
print('Python' and True)
print('Python' or False)
print(False or 'Python')
print(0 and 'Python')
print(0 or False)

True
Python
Python
0
False


연습문제) <br>
표준 입력으로 국어, 영어, 수학, 과학 점수가 입력됩니다. 국어는 90점 이상, 영어는 80점 초과, 수학은 85점 초과, 과학은 80점 이상일 때 합격이라고 정했습니다(한 과목이라도 조건에 만족하지 않으면 불합격). 다음 소스 코드를 완성하여 합격이면 True, 불합격이면 False가 출력되게 만드세요(input에서 안내 문자열은 출력하지 않아야 합니다). <br><br>
입력 (90 81 68 80) <br>
결과 True

In [89]:
Korean, English, Math, Science = map(int,input().split())
print(Korean>=90 and English>80 and Math>85 and Science>=80)

90 81 86 80
True


## **Unit 9. 문자열 사용하기**


' ' (작은 따옴표) <br>
" " (큰 따옴표) <br>
''' (작은 따옴표 3개) <br>
""" (큰 따옴표 3개) <br>

9.1 따옴표 사용법

In [90]:
a = 'Hello, World'
b = '''Hello, Python!'''
c = '''Hello, world!
안녕하세요.
Python입니다.''' # 작은 따옴표 안에 엔터 키를 누르면 다음 줄로 이동함 
print(a)
print(b)
print(c)

Hello, World
Hello, Python!
Hello, world!
안녕하세요.
Python입니다.


In [93]:
s = "Pythong isn't difficult" # 같은 따옴표를 중복으로 사용할 수 없음
r = 'He said "Pythong is easy"'
y = 'Python isn\'t difficult' # 같은 따옴표를 사용하려면 \를 사용하면 됨
print(s)
print(r)
print(y)

Pythong isn't difficult
He said "Pythong is easy"
Python isn't difficult


In [94]:
print('Hello\nPython') # 개행 문자(\n)을 넣으면 따옴표 세 개로 묶지 않아도 여러 줄로 된 문자열을 사용할 수 있음

Hello
Python


연습문제) <br>
다음 소스 코드를 완성하여 실행 결과대로 문자열이 출력되게 만드세요. <br>
'Python' is a "programming language"
that lets you work quickly
and
integrate systems more effectively.

In [99]:
s = ''''Python' is a "programming language" 
that lets you work quickly
and 
integrate systems more effectively.'''
print(s)

'Python' is a "programming language" 
that lets you work quickly
and 
integrate systems more effectively.


## **Unit 10. 리스트와 튜플 사용하기**

10.1
- 리스트 : 값을 일렬로 늘어놓은 형태 <br>
         : [](대괄호) 
         : 각 값은 ,(콤마)로 구분 
         : 리스트 = [값, 값, 값]
         : 모든 자료형 저장 가능

In [100]:
person = ['james', 17, 175.3, True]
person

['james', 17, 175.3, True]

In [102]:
a = [] # 빈 리스트 
b = list() # 빈 리스트 
print(a)
print(b)

[]
[]


10.2 
- 튜플 사용 : 요소를 일렬로 저장, <br>
저장된 요소 변경, 추가, 삭제 불가
            : ()(괄호)
            : 각 값은 ,(콤마)로 구분
            : 괄호 묶지 않고 콤마로만 구분해도 튜플


In [106]:
person = ('james', 17, 175.3, True) # 튜플화 
print(person)
print((98,)) # 한 개짜리 튜플 생성법

('james', 17, 175.3, True)
(98,)


10.3 range 사용 
- range 출력 방식 골라야 함
- range의 마지막 값 ==  입력한 숫자 -1의 값


10.3.1 range를 사용하여 리스트 만들기

In [105]:
print(list(range(10))) # 0부터 10개의 숫자 리스트화 
print(list(range(5,10))) # 시작과 끝 지정해서 리스트화
print(list(range(-4,15,3))) # 시작과 끝 그리고 증가폭 설정한 후 리스트화
print(list(range(10,0,-3))) # 증가폭을 음수로 지정하면 해당 값만큼 숫자 감소 

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
[5, 6, 7, 8, 9]
[-4, -1, 2, 5, 8, 11, 14]
[10, 7, 4, 1]


10.3.2 range를 사용하여 튜프 만들기

In [107]:
print(tuple(range(10)))
print(tuple(range(5,10)))
print(tuple(range(-4,15,2)))


(0, 1, 2, 3, 4, 5, 6, 7, 8, 9)
(5, 6, 7, 8, 9)
(-4, -2, 0, 2, 4, 6, 8, 10, 12, 14)


10.4 튜플 → 리스트 / 리스트 → 튜플

In [108]:
a = [1,2,3]
tuple(a)

(1, 2, 3)

In [109]:
b = (4,5,6)
list(b)

[4, 5, 6]

In [110]:
print(list('Hello')) # 문자 리스트 생성
print(tuple('Hello')) #  문자 튜플 생성

['H', 'e', 'l', 'l', 'o']
('H', 'e', 'l', 'l', 'o')


In [111]:
a, b, c = [1,2,3] # 언패킹
print(a,b,c)

d, e, f = (4,5,6)
print(d,e,f) 

k = (4,5,6) # 튜플 패킹
i,g,t = k
print(i,g,t)

1 2 3
4 5 6
4 5 6


In [112]:
x = input().split()
a, b = x
print(a, b)

50 6
50 6


연습문제) <br>
표준 입력으로 정수가 입력됩니다. range의 시작하는 숫자는 -10, 끝나는 숫자는 10이며 입력된 정수만큼 증가하는 숫자가 들어가도록 튜플을 만들고, 해당 튜플을 출력하는 프로그램을 만드세요(input에서 안내 문자열은 출력하지 않아야 합니다). <br>
입력(2)

In [131]:
k = int(input())
tuple(range(-10,10,k))



2


(-10, -8, -6, -4, -2, 0, 2, 4, 6, 8)

### **Unit 11. 시퀀스 자료형 활용하기**


시퀀스 자료형 : 값이 연속적으로 이어진 자료형 <br>
ex) 리스트, 튜플, range, 문자열, bytes, bytearray <br><br>

요소 : 시퀀스 객체에 들어있는 각 값

11.1 객체에 요소 포함 여부 확인

In [134]:
a = [0, 10, 20, 30, 40, 50, 60]
print(30 in a)
print(100 in a) 
print(100 not in a)
print(30 not in a)

True
False
True
False


In [135]:
'P' in 'Python'

True

11.2 시퀀스 객체 연결

In [136]:
a = [0, 10, 20, 30]
b = [9, 8, 7, 6]
a+b

[0, 10, 20, 30, 9, 8, 7, 6]

In [138]:
range(0, 10) + range(10, 20) # range는 + 연산자로 객체를 연결할 수 없음 
                             # 리스트 또는 튜플로 만들면 연결 가능

TypeError: ignored

In [143]:
print('Hello, ' + 'Python') # 문자 연결
print('Hello, ' + str(10)) # 숫자를 문자화시키면 가능
print('Hello, ' + 10) # 문자와 숫자 연결하면 에러 발생

Hello, Python
Hello, 10


TypeError: ignored

11.3 시퀀스 객체 반복

In [145]:
print([0, 10, 20, 30] * 3)
print(list(range(0, 5, 2)) * 3)
print('Hello, ' * 3)

[0, 10, 20, 30, 0, 10, 20, 30, 0, 10, 20, 30]
[0, 2, 4, 0, 2, 4, 0, 2, 4]
Hello, Hello, Hello, 


11.4 시퀀스 요소 개수 구하기

In [2]:
a = [0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
print(len(a)) # 리스트 요소 개수

b = (38, 76, 43, 62, 19)
print(len(b)) # 튜플 요소 개수

print(len(range(0, 10, 2))) # range 요소 개수

hello = 'Hello, World' # 문자열의 길이
len(hello) # 공백 포함해서 카운트/ 문자열 바이트는 한글 글자 하나 당 3바이트

10
5
5


12

11.5 인덱스 사용

In [19]:
a = [9,5,8,6,2] # 리스트
print(a[0]) # 첫 번째 요소 출력
print(a[-1]) # 마지막 요소 출력
print(a.__getitem__(2)) # 시퀀스객체.__getitem__(인덱스)

b = (5,8,9,7,8) # 튜플
print(b[4]) # 네 번째 요소 출력
print(b.__getitem__(1)) # 두 분째 요소 출력

r = range(2,18,3) # range
print(r[5]) # 다섯 번째 요소 출력
print(r.__getitem__(-3))

h = 'hi, there' # 문자열
print(h[-3]) # 뒤에서 세 번째 요소 출력

p = ['pineapple',2,True,'k'] # 리스트
print(p) # 모든 인덱스 요소 출력
print(p.__getitem__(2)) 

9
2
8
8
8
17
11
e
['pineapple', 2, True, 'k']
True


11.6 마지막 요소에 접근

In [20]:
a = [58,9,8,7,2,33,56]
print(len(a)) # 리스트의 길이 구함
print(a[6])
print(a[len(a)-1]) # 마지막 요소 출력

7
56
56


11.7 요소에 값 할당 <br>
🤯 튜플, range, 문자열: read only

In [27]:
a = [None,None,None,None] # 리스트 
a[0]='hi' # 첫 번째 요소에 값 할당
a[1]=82 # 두 번째 요소에 값 할당
a[2]='bye'
a[3]=99
print(a) # 범위를 벗어난 인덱스 지정 불가

b = (0,0,0,0,0,0) # 튜플
b[0]=22 # 튜플 삭제, 변경, 추가 불가 그러므로 에러 발생
print(b)

c = range(0,12,2) # range
c[2] = 3 # range와 문자열 역시 저장된 요소 변경 불가

['hi', 82, 'bye', 99]


TypeError: ignored

11.8 del로 요소 삭제

In [28]:
a = [98,57,88,65,0] # 리스트
del(a[2]) # 세 번째 요소 삭제
print(a)

b = (38, 21, 53, 62, 19) # 튜플 read only 🤯
del b[2] 
print(b)

r = range(0, 10, 2) # range read only 🤯
del r[2]

hello = 'Hello, world!' # 문자열 read only 🤯
del hello[2]

[98, 57, 65, 0]


TypeError: ignored

11.9 슬라이스 사용

In [3]:
a = [0, 10, 20, 30, 40, 50, 60, 70, 80, 90] # 리스트
print(a[0:4]) # 0 ~ 3번째만 추출
print(a[0:10]) # 전체 출력
print(a[1:1]) # 1 ~ 0번째 추출 → 시작과 끝 인덱스가 같을 시 아무것도 가져오지 않음
print(a[4:-1]) # 음수를 인덱스 지정가능
print(a[2:8:3]) # 슬라이스 안에 증가폭 설정 가능
print(a[:5]) # 인덱스 생략 가능
print(a[:]) # 둘 다 생략 == 전체
print(a[::])
print(a[:5:2]) # 생략하면서 증가폭 설정
print(a[1::3]) # 생략하면서 증가폭 설정
print(a[5:1:-1]) # 음수 폭 지정 가능
print(a[0:len(a)]) # len응용하여 리스트 전체 호출 가능

r = range(10)
print(r[:7:2])
print(list(r[:7:3]))

hello = 'Hello, Python'
print(hello[:6:2])

[0, 10, 20, 30]
[0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
[]
[40, 50, 60, 70, 80]
[20, 50]
[0, 10, 20, 30, 40]
[0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
[0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
[0, 20, 40]
[10, 40, 70]
[50, 40, 30, 20]
[0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
range(0, 7, 2)
[0, 3, 6]
Hlo


11.10 slice 객체 사용

- 슬라이스객체 = slice(끝인덱스)

- 슬라이스객체 = slice(시작인덱스, 끝인덱스)

- 슬라이스객체 = slice(시작인덱스, 끝인덱스, 인덱스증가폭)

- 시퀀스객체[슬라이스객체]

- 시퀀스객체.__getitem__(슬라이스객체)

In [5]:
print(range(10)[slice(4,7,2)])
print(range(10).__getitem__(slice(4,7,2)))

range(4, 7, 2)
range(4, 7, 2)


11.11 슬라이스에 요소 할당
<br>
튜플, range, 문자열 🤯 read only 

In [6]:
a = [10,5,3,6,8,4,5,3,1]
a[2:5] = ['apple','orange','grapefruit']
print(a)

[10, 5, 'apple', 'orange', 'grapefruit', 4, 5, 3, 1]


In [10]:
a = [0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
a[2:5] = 'misletoe' # 슬라이스 범위보다 할당할 요소 개수가 많을 시
print(a)

a[8:13] ='K' # 슬라이스 범위보다 할당할 요소 개수가 적을 시
print(a)

a[2:5:2] = [5,9] # 인덱스 증가폭을 지정하여 요소 할당 🤍 요소 개수 일치
print(a)

[0, 10, 'm', 'i', 's', 'l', 'e', 't', 'o', 'e', 50, 60, 70, 80, 90]
[0, 10, 'm', 'i', 's', 'l', 'e', 't', 'K', 80, 90]
[0, 10, 5, 'i', 9, 'l', 'e', 't', 'K', 80, 90]


11.12 del로 슬라이스 삭제

In [12]:
a = [0, 10, 20, 30, 40, 50, 60, 70, 80, 90]
del a[2:5]
print(a)

del a[2:5:2]
print(a)

[0, 10, 50, 60, 70, 80, 90]
[0, 10, 60, 80, 90]


연습문제) <br>
표준 입력으로 문자열 두 개가 각 줄에 입력됩니다(문자열의 길이는 정해져 있지 않음). 첫 번째 문자열에서 인덱스가 홀수인 문자와 두 번째 문자열에서 인덱스가 짝수인 문자를 연결하여 출력하는 프로그램을 만드세요(input에서 안내 문자열은 출력하지 않아야 합니다). 연결 순서는 첫 번째 문자열, 두 번째 문자열 순입니다. 그리고 0은 짝수로 처리합니다. <br><br>

입력(python<br>
    python) <br>
결과(yhnpto)

In [16]:
a = input()
b = input()
print(a[1::2]+b[::2])

apple
strawberry
plsrwer
